doc/xplat_cli_migration.md

```diff
@@ -0,0 +1,223 @@
+
+1. Switch `azure` commands to `az` commands
+2. Update commands to use new input values
+3. Update script to use new output formats
+4. Use `--query` to extract values reliably
+
+Below, we break down each of these steps.
+
+### Finding and switching to `az` commands
+
+While most commands keep the same group and command names between the Azure XPlat CLI and the Azure CLI 2.0, we've built a [azure to az conversion table](https://github.com/Azure/azure-cli/blob/master/doc/azure2az_commands.rst) for common commands.
+
+#### Set vs. Update
+
+Mutate operations now use the `update` verb instead of `set`.  While the XPlat CLI
+exposed some common operations as parameters, such as:
+
+```
+$ azure vm set -g MyGroup -n MyName --nic-ids $MyNicID
+$ azure vm set -g MyGroup -n MyName --tags myTagName=MyTagValue
+```
+
+The Azure CLI 2.0 `update` commands work generically against the resource, for example:
+```
+$ az vm update -g MyGroup -n MyName --add networkProfile.networkInterfaces primary=false id=$MyNicID
+$ az vm update -g MyGroup -n MyName --set tags.myTagName=MyTagValue
+```
+
+#### Commands with complex input
+
+Some commands which required complex input, such as JSON strings and documents,
+have been changed to add/remove single values, instead of requiring users to
+pull down the document, modify, and set the new document.  For these commands,
+see help (`-h`) for details on the specific command.
+
+An example of this is `azure storage cors set` being replaced by `az storage cors add`.
+
+### Updating input values
+
+Once you have identified the `az` commands required for your script, you immediately
+notice changes to how inputs are handled.  The Azure CLI 2.0 does not accept
+'positional parameters' such as `azure vm show MyRG MyName`, but instead require
+parameter flags: `az vm show -g MyRG -n MyName`.  
+
+In addition, when an input value is missing, we will show an error indicating the
+missing parameters, instead of prompting the user automatically:
+
+```
+$ az vm show
+az vm show: error: (--name --resource-group | --ids) are required
+```
+
+In addition to use resource groups and names (`-g -n`), you can also refer to
+resources directly by ID value using `--ids`:
+
+```
+$ MyVar=$(az vm list --query [0].id --out tsv)
+$ echo $MyVar
+/subscriptions/db5eabce-73e2-4fa8-b18a-46abcbe4abc5/resourceGroups/VMGROUP1/providers/Microsoft.Compute/virtualMachines/VM-Data
+$ az vm show --ids $MyVar
+ResourceGroup    Name     VmId                                  Location    ProvisioningState
+---------------  -------  ------------------------------------  ----------  -------------------
+VMGROUP1         VM-Data  63edd6a0-2796-49e6-acc1-ad3f8bd94f13  westus      Succeeded
+```
+
+### Working with output formats
+
+The Azure CLI 2.0 supports 4 primary output formats:
+
+1. json  - standard JSON formatted object graphs
+2. jsonc - colorized JSON
+3. tsv   - provides UNIX-style output
```

Comment:
>UNIX-style output [](start = 20, length = 17)

not sure this is an accurate way to describe tsv, it might be better to just call out that it is tab separated values